generator/src/lib.rs

```diff
@@ -1613,7 +1612,9 @@ pub fn derive_setters(
     // implement
     let next_trait = if has_next && _struct.extends.is_none() {
         quote! {
-            pub unsafe trait #extends_name {}
+            pub unsafe trait #extends_name {
+                unsafe fn as_ptr_mut(&mut self) -> *mut BaseOutStructure;
```

Comment:
Instead of generating more code here, we should mark all builders as `#[repr(transparent)]` and coerce the pointers directly in `push_next`. This is legal because the builders contain only one non-zero-sized member.

Reply:
That is a great idea, I completely forgot about `#[repr(transparent)]`.